File: database/src/main/java/it/niedermann/nextcloud/deck/database/dao/ActivityDao.java
package it.niedermann.nextcloud.deck.database.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Query;

import java.util.List;

import it.niedermann.nextcloud.deck.model.ocs.Activity;

@Dao
public interface ActivityDao extends GenericDao<Activity> {

    @Query("SELECT * FROM activity WHERE cardId = :localCardId order by lastModified desc")
    LiveData<List<Activity>> getActivitiesForCard(final long localCardId);

    @Query("SELECT * FROM activity WHERE accountId = :accountId and id = :remoteActivityId")
    Activity getActivityByRemoteIdDirectly(long accountId, long remoteActivityId);
}

File: data/local/src/main/java/it/niedermann/nextcloud/deck/data/local/dao/ActivityDao.kt
package it.niedermann.nextcloud.deck.data.local.dao

import androidx.room3.Dao
import androidx.room3.Insert
import androidx.room3.OnConflictStrategy
import androidx.room3.Query
import io.reactivex.rxjava3.core.Flowable
import it.niedermann.nextcloud.deck.data.local.entity.ActivityEntity
import java.util.concurrent.CompletableFuture

@Dao
interface ActivityDao : GenericDao<ActivityEntity> {

    @Query("SELECT * FROM Activity WHERE cardId = :cardId")
    fun getActivitiesByCard(cardId: Long): Flowable<List<ActivityEntity>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertOrReplace(entity: ActivityEntity): CompletableFuture<Long>
}
